rfp-proposals/rfp-proposal-reputation-system.md

```diff
@@ -0,0 +1,326 @@
+    * Miner Proactiveness: 
+        * Delta of time since Filecoin published an upgrade for node software and the miner updated itself. 
+        * Reason: 
+            * Smaller the delta, the more proactive the miner. 
+            * Miners who don't update are risky. 
+
+
+   
+### Specifics of web app: 
+
+
+#### Home page:
+Displaying  list of miners currently providing space for allocation with the following info:
+* Miner
+* Miner Score
+* Used Storage
+* Available storage
+* Storage Price
+* Valid till
+* Status
+
+ 
+![](https://i.imgur.com/OsVK6v2.png)
+
+^ Early Prototype. 
+
+<br><br>
+
+#### Miner Individual page:
+* Miner stats card containing information:
+    * Miner Address
+    * Filecoin version
+    * Collateral 
+    * Miner geographic location
+    * Storage Metrics: Used/Available/Total Storage. 
+
+* Miner Score Table: (10 days average):
+    * [Final/Average Score](#How-will-scores-be-calculated?)
+    * Individual characteristics score:
+        * Latency
+        * % of successful deals
+        *  %  of slashed and dropped contracts
+        * etc
+
+* Comparison with storage miners with the same score:
+    * Storage price
+    * Collateral
+    * Est. uptime
+    * Stored data
+
+* Substantiated feedback:
+    * identifying the user:
+        * a message-signing-based authentication mechanism with a user's public address as their identifier.
+    * identifying the event:
+        * enter dealid/(id associated with the event): 
+        * Choose check event:
+        * Comment/feedback
+        * Submit
+        * Post if the event is verified.
+
+
+![](https://i.imgur.com/zDz75Bd.png)
+
+^ Early Prototype. 
+
+
+
+## Development Roadmap
+
+Tentative Start: Feb 1, 2019
+### Milestone 1: Get vantage on: 
+
+1. Scope of the reputation system service: 
+    * Collaborate with FileCoin DevGrants team to: 
+        * Gain feedback on the metrics we have thought of. How relevant would they be for users now?  
+        * Gain awareness on the kinds of objective feedback users will give that can be verifiable from the service.
+    * Iron out reputation score logic that will be the single metric that all miners will be ranked upon. 
+    * Finalize the reputation system service requirements spec sheet.
+     
+2. and how the tool would be built: 
+    * Syncup process from Filecoin Network to DB. 
+    * Architectural Design vantage of the service.   
+
+People Involved: 
+1. Lead Developer. 
+2. Filecoin Researcher. 
+3. Devops Engineer. 
+
+Milestone Completion ETA: 15 Feb, 2020
+Duration: 2 weeks. 
+Funding Required: $10,000 
+
+
+### Milestone 2: #buidl-ing the service 
+
+1. Setting up the sync progress from Filecoin Node to DB. 
+2. Writing API's + code for users to: 
+    * query miner metrics above.
+    * submit claims: 
+        * that get verified via data that is contextual to the claim. 
```

Comment:
Can you provide an example  of a claim that gets verified?

Reply:
Sure. One of the claims by clients could be that data was removed before the deal was supposed to expire. This is our speculation so far from my research that if a client loses data that they had stored with a particular miner, even though the protocol will slash the miner, the client may not know about the fact that the miner was slashed. This is how I imagine the service to work: 

1. Client submits claim.  
2. Claim get’s verified. Appropriate info is shown to the user. 
3. Once the claim is verified -> they can use the link to share it with the community. 

Additional context: As mentioned in milestone 1 of our proposal, our team will be deep diving into ConsensusFaults and StorageDealFaults to figure out possible claims clients can raise which can be verified from on chain activity.